de/nmichael/efa/data/storage/Journal.java

```diff
@@ -316,12 +316,14 @@ public boolean log(long scn, Operation operation, DataRecord r) {
             if (s == null || s.length() == 0) {
                 Logger.log(Logger.ERROR, Logger.MSG_DATA_JOURNALWRITEFAILED,
                         LogString.fileWritingFailed(fwname, International.getString("Journal"), "empty log string"));
+                f.close(); // close the writer to avoid resource leak
                 return false;
             }
             f.write(s + "\n");
-            if (FLUSH_WRITES) {
+            if (FLUSH_WRITES) { //TODO: does this make sense here? the buffered writer will be closed (and thus flushed) as it is a local variable..
```

Comment:
This does make sense (see above). We're keeping the journal open but just flushing it here.
f is just a reference to the instance variable fw returned b openForAppen().